kolibri/plugins/coach/assets/src/views/reports/ReportsExerciseLearners.vue

```diff
@@ -0,0 +1,108 @@
+<template>
+
+  <CoreTable :emptyMessage="coachStrings.$tr('activityListEmptyState')">
+    <thead slot="thead">
+      <tr>
+        <th>{{ coachStrings.$tr('nameLabel') }}</th>
+        <th>{{ coachStrings.$tr('progressLabel') }}</th>
+        <th>{{ coachStrings.$tr('timeSpentLabel') }}</th>
+        <th v-if="showGroupsColumn">
+          {{ coachStrings.$tr('groupsLabel') }}
+        </th>
+        <th>{{ coachStrings.$tr('lastActivityLabel') }}</th>
+      </tr>
+    </thead>
+    <transition-group slot="tbody" tag="tbody" name="list">
+      <tr v-for="entry in entries" ref="entry" :key="entry.id">
```

Comment:
Is the `ref` used somewhere outside of this component?

Reply:
Yes, in specs - I usually prefer to use `ref`s there.

Reply:
Which 'tr' does `$refs.entry` refer to? The last one?

In your test is it possible to write the test using something more high-level, like the data from vue-test-utils Wrapper methods, instead of having to access more low-level DOM info from `refs`?

Reply:
Basically, it's used only to find those elements using vue-test-utils `find` - in case there is markup/css selector change, you don't need to update your `find`s, when compared to trying to find an element by a class or tag, that's all. Both ways have some advantages/disadvantages, no problem to change it if you don't use it in the codebase.

Reply:
We generally reserve the use `ref` if we need to access it in JS application code. Even if it's used in a test, it sort of looks like dead code, and I have the tendency to delete `refs` if I see they aren't used anymore in `methods`, etc.

People who want to avoid using "brittle" CSS selectors have been using this test-specific attribute `data-test`. In the spec, you are using `ref` as an attribute selector, and not really using the `ref` in the same way we use it components (e.g. low-level DOM access).

Reply:
Yes, it's one of the disadvantages indeed.

I like the idea of using the attribute, it combines the best of both worlds, thanks for the tip!